softwareComponents/networking/include/physical_netif.hpp

```diff
@@ -133,7 +133,7 @@ class PhysNetif {
     }
 
     static err_t linkOutput( struct netif*, struct pbuf* ) {
-        ROFI_UNREACHABLE( "Link output should not be directly used" );
+        assert( false && "Link output should not be directly used" );
```

Comment:
Tady je na zváženou, jestli ROFI_UNREACHABLE vlastně není správně. Má to za úkol dát programátorovi přes prsty. Ale pak netřeba mít `return ERR_OK`. 

Tady vlastně stálo za to znát názor @jajik

Reply:
No... problém s `ROFI_UNREACHABLE` je, že v debugu se sice chová jako assert, ale v releasu ti dá přes prsty mnohem víc, než bys chtěl (nejspíš úplně jinde), protože to dá optimalizátoru flag, že sem se nikdy nedostane, takže cesty co tam vedou můžeš zahodit...

Reply:
Tak pak je otázka, jestli raději nechceme ještě `ROFI_TRAP`.

Reply:
To určitě můžeme - zvažovali jsme ten ROFI_ASSURE (který i v releasu kontrolu podmínku a případně udělá TRAP)

Reply:
Přidejme TRAP s následujícím chováním:
- v Debug módu to co nejgrandiózněji oznámí, co je špatně (podobně jako assert) a pak se to zasekne.
- v Release módu tu zasekne

Alternativně nad čím přemýšlím: což takto přidat "trap handler". Obyčejná funkce přilinkovaná k projektu, která řeší chování trapu.

Reply:
`__builtin_trap`

Reply:
Povrzení od @jajik: Když toto zavoláš, má to umřít, protože všechno ostatní co následuje poté může být nedefinované.

Reply:
Ona je "virtuální". Ale v Cčku.

Reply:
Vrátil jsem ty změny v `networking`, abychom se někam pohli a vytvořil jsem k tomu issue #250 (a smazal jsem irelevantní return na konci této funkce).

Reply:
Jinak důvod: myslím si, že k tomu budem potřebovat víc funkcionality (maker) v atoms, což je nad scope tohoto PR